src/main/scala/rocket/Instructions.scala

```diff
@@ -319,52 +319,40 @@ object Instructions {
-  def VSW_V              = BitPat("b???000?00000?????110?????0100111")
-  def VSE_V              = BitPat("b???000?00000?????111?????0100111")
-  def VLSB_V             = BitPat("b???110???????????000?????0000111")
-  def VLSH_V             = BitPat("b???110???????????101?????0000111")
-  def VLSW_V             = BitPat("b???110???????????110?????0000111")
-  def VLSE_V             = BitPat("b???010???????????111?????0000111")
-  def VLSBU_V            = BitPat("b???010???????????000?????0000111")
-  def VLSHU_V            = BitPat("b???010???????????101?????0000111")
-  def VLSWU_V            = BitPat("b???010???????????110?????0000111")
-  def VSSB_V             = BitPat("b???010???????????000?????0100111")
-  def VSSH_V             = BitPat("b???010???????????101?????0100111")
-  def VSSW_V             = BitPat("b???010???????????110?????0100111")
-  def VSSE_V             = BitPat("b???010???????????111?????0100111")
-  def VLXB_V             = BitPat("b???111???????????000?????0000111")
-  def VLXH_V             = BitPat("b???111???????????101?????0000111")
-  def VLXW_V             = BitPat("b???111???????????110?????0000111")
-  def VLXE_V             = BitPat("b???011???????????111?????0000111")
-  def VLXBU_V            = BitPat("b???011???????????000?????0000111")
-  def VLXHU_V            = BitPat("b???011???????????101?????0000111")
-  def VLXWU_V            = BitPat("b???011???????????110?????0000111")
-  def VSXB_V             = BitPat("b???011???????????000?????0100111")
-  def VSXH_V             = BitPat("b???011???????????101?????0100111")
-  def VSXW_V             = BitPat("b???011???????????110?????0100111")
-  def VSXE_V             = BitPat("b???011???????????111?????0100111")
-  def VSUXB_V            = BitPat("b000111???????????000?????0100111")
-  def VSUXH_V            = BitPat("b000111???????????101?????0100111")
-  def VSUXW_V            = BitPat("b000111???????????110?????0100111")
-  def VSUXE_V            = BitPat("b000111???????????111?????0100111")
-  def VLBFF_V            = BitPat("b???100?10000?????000?????0000111")
-  def VLHFF_V            = BitPat("b???100?10000?????101?????0000111")
-  def VLWFF_V            = BitPat("b???100?10000?????110?????0000111")
-  def VLEFF_V            = BitPat("b???000?10000?????111?????0000111")
-  def VLBUFF_V           = BitPat("b???000?10000?????000?????0000111")
-  def VLHUFF_V           = BitPat("b???000?10000?????101?????0000111")
-  def VLWUFF_V           = BitPat("b???000?10000?????110?????0000111")
-  def VL1R_V             = BitPat("b000000101000?????111?????0000111")
-  def VS1R_V             = BitPat("b000000101000?????111?????0100111")
+  def VLE8_V             = BitPat("b???000?00000?????000?????0000111")
+  def VLE16_V            = BitPat("b???000?00000?????101?????0000111")
+  def VLE32_V            = BitPat("b???000?00000?????110?????0000111")
+  def VLE64_V            = BitPat("b???000?00000?????111?????0000111")
```

Comment:
Why did you define `VLE...` opcodes for every width instead of single `VLE` opcode and then extracting the width from instruction bits directly?

Reply:
Note all three funct3 bits get used (bit14 is 0 for VLE8, but it’s 1 for the others). This is because FLH/FLW/FLD/FLQ are in the same major opcode. If I replaced funct3 with ???, then VLE would conflict with the FP loads.

Of course this is just for illegal instruction detection. Once you get past that point, I agree you should just extract bits 13:12 to determine the width, rather than treating these as four separate instructions.